File: snippets/fsharp/System/NullReferenceException/Overview/Array1.fs
module Array1

// <Snippet8>
open System

let values = [| "one"; null; "two" |]
for i = 0 to values.GetUpperBound 0 do
    printfn $"""{values[i].Trim()}{if i = values.GetUpperBound 0 then "" else ", "}"""
printfn ""
// The example displays the following output:
//    Unhandled Exception:
//       System.NullReferenceException: Object reference not set to an instance of an object.
//       at <StartupCode$fs>.main()
// </Snippet8>

File: snippets/csharp/System/NullReferenceException/Overview/Array1.cs
using System;

public class Array1Example
{
    public static void Main()
    {
        // <Snippet8>
        string[] values = [ "one", null, "two" ];
        for (int ctr = 0; ctr <= values.GetUpperBound(0); ctr++)
            Console.Write("{0}{1}", values[ctr].Trim(),
                          ctr == values.GetUpperBound(0) ? "" : ", ");
        Console.WriteLine();

        // The example displays the following output:
        //    Unhandled Exception:
        //       System.NullReferenceException: Object reference not set to an instance of an object.
        // </Snippet8>
    }
}
